test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec_swap/src/main.sw

```diff
@@ -0,0 +1,787 @@
+script;
+
+use std::{assert::assert, hash::sha256, option::Option, revert::revert, u128::{From, U128}, vec::Vec};
+
+enum SimpleEnum {
+    A: b256,
+    B: (),
+}
+
+struct SimpleStruct {
+    x: u32,
+    y: b256,
+}
+
+fn main() -> bool {
+    test_vector_swap_u8();
+    test_vector_swap_u128();
+    test_vector_swap_b256();
+    test_vector_swap_struct();
+    test_vector_swap_enum();
+    test_vector_swap_tuple();
+    test_vector_swap_string();
+    test_vector_swap_array();
+    test_vector_swap_same_indexes_noop();
+    true
+}
+
+fn test_vector_swap_u8() {
+    let mut vector = ~Vec::new();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+    let number2 = 2u8;
+
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+
+    assert(vector.len() == 3);
+    assert(vector.capacity() == 4);
+    assert(vector.is_empty() == false);
+
+    match vector.get(0) {
+        Option::Some(val) => {
+            assert(val == number0)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+
+    match vector.get(1) {
+        Option::Some(val) => {
+            assert(val == number1)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+
+    match vector.get(2) {
+        Option::Some(val) => {
+            assert(val == number2)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+
+    vector.swap(0, 2);
+
+    assert(vector.len() == 3);
+    assert(vector.capacity() == 4);
+    assert(vector.is_empty() == false);
+
+    match vector.get(0) {
+        Option::Some(val) => {
+            assert(val == number2)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+
+    match vector.get(1) {
+        Option::Some(val) => {
+            assert(val == number1)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+
+    match vector.get(2) {
+        Option::Some(val) => {
+            assert(val == number0)
+        },
+        Option::None => {
+            revert(0)
+        },
+    }
+}
+
+fn test_vector_swap_u128() {
```

Comment:
I wouldn't bother testing custom types like this. Lots will be added over time and this is just a repeat of a struct test

Reply:
I removed test_vector_swap_u128 now.